src/v/cloud_storage/tests/remote_segment_test.cc

```diff
@@ -55,6 +55,19 @@ using namespace cloud_storage;
 
 inline ss::logger test_log("test"); // NOLINT
 
+/**
+ * Helper: generate a function suitable for passing to upload_segment(),
+ * exposing some synthetic data as a segment_reader_handle.
+ */
+std::function<ss::future<storage::segment_reader_handle>()>
```

Comment:
noncopyable_fn? do you need copy?

Reply:
This was a preexisting interface that took a std::function, but while we're here let's clean up -- fixed.